### ISSUE TITLE
[GNU social] Available character count is not correct

### ISSUE THREAD

**Issue by benediktg** [TwidereProject/Twidere-Android#481]
Since the number of available character should now be guessed automatically (kudos for that effort) I have noticed that with gnusocial.de the character count is wrong. It should be 1024 [1] but actually I get only 140 characters [2].

FYI, I have installed Twidere v3.1.0.9 from F-Droid and just tell me if you need more info.

[1]:

```
curl -s https://gnusocial.de/api/statusnet/config.json | tr ',' '\n' | grep textlimit
```

returns

```
"textlimit":"1024"
```

[2]:

![screenshot_20160324-184209](https://cloud.githubusercontent.com/assets/14795615/14025834/4c092386-f1f0-11e5-8957-0478f7befa85.png)

also concerns #296

**Comment by annando** [comment]
I guess that the limit is fetched when the account was created. There doesn't seem to be an option to refetch the limit.

**Comment by mariotaku** [comment]
@annando Yep, that's how text limit works.

@benediktg Maybe you could try re-add your account?

**Comment by dtluna** [comment]
![2016-03-27 14 28 13](https://cloud.githubusercontent.com/assets/10116666/14065221/8bc80b04-f42a-11e5-8245-7d9e6465ce68.png)
On gs.smuglo.li which has no character limit, I can't send anything.

**Comment by benediktg** [comment]
@mariotaku Thanks, re-adding the account worked so that I got a message that I am already logged in. After that the right chararacter count was fetched.

I think you just need to interpret the ```"textlimit":"0"``` as unlimited.

**Comment by mariotaku** [comment]
Fixed in fb9969f98b078f6043b8c47f6f16ecf883d78db3.
